packages/big-design/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo, useReducer } from 'react';
+
+import { typedMemo } from '../../utils';
+
+import { createReducer, createReducerInit } from './reducer';
+import { StyledUl } from './styled';
+import { TreeNode } from './TreeNode';
+import { TreeProps } from './types';
+
+const InternalTree = <T extends unknown>({
+  iconless,
+  nodes,
+  onCollapse,
+  onExpand,
+  onSelect,
+  selectable,
+}: TreeProps<T>): React.ReactElement<TreeProps<T>> => {
+  const reducer = useMemo(() => createReducer<T>(), []);
+  const reducerInit = useMemo(() => createReducerInit<T>(), []);
+  const [state, dispatch] = useReducer(reducer, { nodes, radio: selectable === 'radio' }, reducerInit);
+
+  const renderedItems = useMemo(
+    () =>
+      nodes.map((node, index) => (
+        <TreeNode
+          {...node}
+          dispatch={dispatch}
+          iconless={iconless}
+          key={index}
+          onCollapse={onCollapse}
+          onExpand={onExpand}
+          onSelect={onSelect}
+          selectable={selectable}
+          state={state}
+        />
+      )),
+    [iconless, nodes, onCollapse, onExpand, onSelect, selectable, state],
+  );
+
+  return (
+    <StyledUl role="tree" aria-multiselectable={selectable === 'multi'} style={{ overflow: 'hidden' }}>
```

Comment:
Why have the `overflow: hidden` as an inline style?

Reply:
Good question! I only need the overflow on that one `ul` so I opted in for this instead of creating a whole new SC for it. I tried to move it into the SC, but there's some weird nesting stuff that causes styles to override the parent styles and they don't need to be complex.